### ISSUE TITLE
BMR3, enemy and allied flags are invisible

### ISSUE THREAD

**Issue by Konrad22** [doofus-01/wesnoth-Bad_Moon_Rising#6]
![alliedflag](https://user-images.githubusercontent.com/39015360/54878080-9ad74100-4e27-11e9-9921-60dad3d92407.png)
![enemyflag](https://user-images.githubusercontent.com/39015360/54878082-9e6ac800-4e27-11e9-8828-ec3ac878665f.png)

**Comment by doofus-01** [comment]
I can't reproduce this issue, in the first couple turns at least.

**Comment by Konrad22** [comment]
A replay for now, to check if you can observe the issue. I'll try figuring out when this started happening in the meantime.
[BMR3-Escape the Legion replay.gz](https://github.com/doofus-01/wesnoth-Bad_Moon_Rising/files/3000940/BMR3-Escape.the.Legion.replay.gz)

Update:
So, this started happening in 'Mine Mouth'.
[BMR3-Mine Mouth.gz](https://github.com/doofus-01/wesnoth-Bad_Moon_Rising/files/3000943/BMR3-Mine.Mouth.gz)
[BMR3-Mine Mouth replay.gz](https://github.com/doofus-01/wesnoth-Bad_Moon_Rising/files/3000944/BMR3-Mine.Mouth.replay.gz)

The scenario start file is from 1.14.5 but I probably loaded it in 1.14.6. Could that be a cause?

**Comment by doofus-01** [comment]
Yes, sort of.  There was clean-up of some orcish flags in the campaign, so old files will call for a set of images that no longer exist.  New saves should be OK.
This is the same issue as #4 , I'll close that one.

**Comment by Konrad22** [comment]
New saves as in 'saves from the campaign started in 1.14.6' or 'saves from scenarios started after 1.14.6'? Because the latter is not true.

EDIT:
If you were referring to the version number of BMR, that one is constant 1.10.4 for all of my BMR3 saves.

**Comment by doofus-01** [comment]
The flag images were in the campaign (they were there before they got in mainline), I don't think there's anything special about BfW 1.14.6 versus 1.14.5 in this regard.  If the save-files all come from the same campaign version, that is more puzzling...

Where this reference to the old image files is coming from isn't clear to me right now.  I grep'd the campaign files for them, and don't come up with anything active, and it doesn't happen when I test from scenario start.

**Comment by Konrad22** [comment]
I _might_ have figured it out.
First things first, my assumptions rely on:
a) You really didn't use orcish flags for the orcs in BMR2 - A New Home and BMR3 - Orcish Forces.
b) Starting BMR3 from the campaign menu after finishing BMR2 makes the BMR3 saves inherit some 'properties' from the (last?) BMR2 saves, e.g. the orcish flags it calls upon.

It has probably nothing to do with 1.14.5 vs 1.14.6, that's just a concidence. The reason why I only noticed I ran into that bug only at 'Mine Mouth' was because you didn't use orcish flags before in BMR3 (not even in 'Orcish Skirmish' which is something improveable). 
b) is important, because I was playing Ruin with 1.10.3 which would influence my save files for BMR2 to come, giving them the 'property' of calling onto that old set of images. Which was inherited by BMR3 in turn.

EDIT:
Almost forgot to mention that I can reproduce this bug when starting BMR3 from the campaign list, so it seems to make sense.

**Comment by doofus-01** [comment]
I searched through my persistence file (located in `<wesnoth_user_data>'/persist/Bad_Moon_Rising.cfg`) and didn't come up with the image names.  That file was last modified on January 6th.  I guess if yours is older, or has an image with "orc-flag" referenced in it, that could explain it.

**Comment by Konrad22** [comment]
Mine was modifed 6 days ago. :(
